lib/matplotlib/colorbar.py

```diff
@@ -852,12 +979,28 @@ def _mesh(self):
             y = self._uniform_y(self._central_N())
         else:
             y = self._proportional_y()
+        # if boundaries and values are None, then we can go ahead and
+        # scale this up for Auto tick location.  Otherwise we
+        # want to keep normalized between 0 and 1 and use manual tick
+        # locations.
+        if self._use_adjustable():
+            y = self.norm.inverse(y)
+            x = self.norm.inverse(x)
+        else:
+            dy = 1.0
         self._y = y
+
         X, Y = np.meshgrid(x, y)
         if self._extend_lower() and not self.extendrect:
-            X[0, :] = 0.5
+            if self._use_adjustable():
```

Comment:
This can also be condensed:
```python
xmid = self.norm.inverse(0.5) if self._use_adjustable() else 0.5
if self._extend_lower() and not self.extendrect:
    X[0, :] = xmid
if self._extend_upper() and not self.extendrect:
    X[-1, :] = xmid
```